docs/src/main/asciidoc/_high-availability/cloud/Kubernetes.adoc

```diff
@@ -0,0 +1,230 @@
+In order to achieve optimal performance, the load balancer deployed in front of the application server cluster:
+
+1. must route HTTP requests for a given session to the correct primary owner,
+2. as the topology changes, the application server must be able to update the affinity information.
+
+This can be done by configuring HAProxy Ingress Controller described below.
+
+=== Changing Cache Mode in Deployment Image
+
+// TODO - update the wording when we make 'dist' the actual cloud default - https://issues.redhat.com/browse/CLOUD-4211
+Presently, existing applications need to update the default cache mode to optimize scalability.
+
+Applications built using Maven with https://docs.wildfly.org/wildfly-maven-plugin/releases/4.2/package-mojo.html#packagingScripts[WildFly Maven Plugin]
+to create an image can be easily configured to update the default cache mode.
+This can be done by configuring a set commands in the packaging step to reconfigure the bundled server.
+The following `infinispan.cli` configuration script will reconfigure the `infinispan` subsystem to use the distributed cache for web sessions:
+
+// n.b. regarding the recommended configuration:
+// we are intentionally leave out the file store (spin a new instance when needed) + leave out locking and transactions.
+// The provided expiration configuration saves unnecessary 1 expiration thread that would otherwise run.
+
+[source,cli,title=infinispan.cli]
+----
+/subsystem=infinispan/cache-container=web/distributed-cache=sessions:add
+/subsystem=infinispan/cache-container=web/distributed-cache=sessions/component=expiration:add(interval=0)
+/subsystem=infinispan/cache-container=web:write-attribute(name=default-cache,value=sessions)
+----
+
+The Maven plugin configuration can then be updated to update the generated container image:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-maven-plugin</artifactId>
+    <executions>
+        <execution>
+            <!-- ... -->
+            <configuration>
+                <!-- ... -->
+                <packaging-scripts>
+                    <packaging-script>
+                        <scripts>
+                            <script>infinispan.cli</script>
+                        </scripts>
+                    </packaging-script>
+                </packaging-scripts>
+            </configuration>
+        </execution>
+    </executions>
+</plugin>
+----
+
+== Ingress Configuration
+
+A https://kubernetes.io/docs/concepts/services-networking/ingress/[Kubernetes Ingress] is an object that manages external access to the services in a cluster, typically via HTTP.
+They provide load balancing, SSL termination and name-based virtual hosting.
+An Ingress Controller implementation supporting L7 load-balancing that supports the above affinity requirements will need to be configured.
```

Comment:
In order to support the affinity requirements described above, an Ingress Controller implementation, supporting L7 load-balancing and exposes the requisite annotations for configuring session affinity, will need to be configured.

Reply:
Reworded.